server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -400,24 +405,34 @@ private int write(final Collection<CommitSession> sessions, boolean mincFinish,
         if (currentLogId == logId.get()) {
 
           // write the mutation to the logs
-          seq = seqGen.incrementAndGet();
-          if (seq < 0)
-            throw new RuntimeException("Logger sequence generator wrapped!  Onos!!!11!eleven");
-          LoggerOperation lop = writer.write(copy, seq);
+          LoggerOperation lop = writer.write(copy);
           lop.await();
 
           // double-check: did the log set change?
           success = (currentLogId == logId.get());
         }
       } catch (DfsLogger.LogClosedException ex) {
-        log.debug("Logs closed while writing, retrying " + attempt);
+        log.debug("Logs closed while writing, retrying " + writeRetry.retriesCompleted());
       } catch (Exception t) {
-        if (attempt != 1) {
-          log.error("Unexpected error writing to log, retrying attempt " + attempt, t);
+        // We have more retries or we exceeded the maximum number of accepted failures
+        if (writeRetry.canRetry()) {
+          // Use the createRetry and record the time in which we did so
+          writeRetry.useRetry();
+
+          try {
+            // Backoff
+            writeRetry.waitForNextAttempt();
+          } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+          }
+        } else {
+          log.error("Repeatedly failed to write WAL. Going to exit tabletserver.", t);
```

Comment:
Is a termination reasonable in this circumstance or should we retry forever as we were doing before?